pkg/scheduler/priorities.go

```diff
@@ -132,3 +134,92 @@ func (n *NodeLabelPrioritizer) CalculateNodeLabelPriority(pod api.Pod, podLister
 	}
 	return result, nil
 }
+
+// BalancedResourceAllocation is favors nodes with balanced resource usage rate.
+// BalancedResourceAllocation should **NOT** be used alone, and **MUST** be used together with LeastRequestedPriority.
+// It calculates the percentage of memory and CPU requested over capacity, and prioritizes based on how close the
+// two percentage metrics are to each other.
+// Details: score = 1 / sqrt[(cpu_pct - average)^2 + (mem_pct - average)^2], average = (cpu_pct + mem_pct) / 2
+func BalancedResourceAllocation(pod api.Pod, podLister PodLister, minionLister MinionLister) (HostPriorityList, error) {
+	nodes, err := minionLister.List()
+	if err != nil {
+		return HostPriorityList{}, err
+	}
+	podsToMachines, err := MapPodsToMachines(podLister)
+
+	list := HostPriorityList{}
+	for _, node := range nodes.Items {
+		list = append(list, calculateBalancedResourceAllocation(pod, node, podsToMachines[node.Name]))
+	}
+	return list, nil
+}
+
+func calculateBalancedResourceAllocation(pod api.Pod, node api.Node, pods []api.Pod) HostPriority {
+	totalMilliCPU := int64(0)
+	totalMemory := int64(0)
+	score := int(0)
+	for _, existingPod := range pods {
+		for _, container := range existingPod.Spec.Containers {
+			totalMilliCPU += container.Resources.Limits.Cpu().MilliValue()
+			totalMemory += container.Resources.Limits.Memory().Value()
+		}
+	}
+	// Add the resources requested by the current pod being scheduled.
+	// This also helps differentiate between differently sized, but empty, minions.
+	for _, container := range pod.Spec.Containers {
+		totalMilliCPU += container.Resources.Limits.Cpu().MilliValue()
+		totalMemory += container.Resources.Limits.Memory().Value()
+	}
+
+	capacityMilliCPU := node.Status.Capacity.Cpu().MilliValue()
+	capacityMemory := node.Status.Capacity.Memory().Value()
+
+	cpuPctOfCapRequested := calculatePctOfCapRequested(totalMilliCPU, capacityMilliCPU, node.Name)
+	memoryPctOfCapRequested := calculatePctOfCapRequested(totalMemory, capacityMemory, node.Name)
+	avergePctOfCapRequested := (cpuPctOfCapRequested + memoryPctOfCapRequested) / 2
+	cpuDiff := cpuPctOfCapRequested - avergePctOfCapRequested
+	memoryDiff := memoryPctOfCapRequested - avergePctOfCapRequested
+	diff := math.Sqrt(math.Pow(cpuDiff, 2) + math.Pow(memoryDiff, 2))
+	// closer percentage metrics lead to higher score with 10 being the highest
```

Comment:
I'm not really sure about this scaling mechanism of mathematical inverse here, it seems ad-hoc?

Compare: https://www.wolframalpha.com/input/?i=graph+sqrt%28x*x+%2B+y*y%29

vs: https://www.wolframalpha.com/input/?i=graph+1%2Fsqrt%28x*x+%2B+y*y%29

The former looks much more gentle than the latter, no?

What do you think about:
a) compute the upper and lower bound of diff
b) scale to the range 0-10
c) then subtract this from 10 instead of doing the inverse.

Or maybe I'm missing something important?